docs/v1/api.md

```diff
@@ -0,0 +1,230 @@
+# Service Catalog API
+
+This document proposes a Kubernetes-native service broker API. It is a formalization of [@krancour](http://github.com/krancour)'s [presentation](https://docs.google.com/presentation/d/1Fm3qG9zQ4R8ZbhtTObmQjO3UZYJ0ruYI4qr5w_gEKN8/edit?usp=sharing) that he gave at the sig-service-catalog meeting on 10/10/2016.
```

Comment:
@pmorie thoughts on making it the design document in this PR? I'd only need to change some wording to do so